assets/blocks/course-outline/data.js

```diff
@@ -71,26 +71,51 @@ export const syncStructureToBlocks = ( structure, blocks ) => {
 	} );
 };
 
+/**
+ * Predicate method to compare a block with a course structure item, using different strategies.
+ *
+ * @param {Object[]}                                    blocks        Block.
+ * @param {Array.<(CourseLessonData|CourseModuleData)>} structureItem Structure item.
+ * @return {boolean}  Flag indicating if the block matches on the the strategies
+ */
+
+const byStructureItem = ( block ) => ( courseStructureItem ) => {
+	const { name, attributes } = courseStructureItem;
```

Comment:
Nicely done 👍 Just a small issue here probably, not sure, but should it be like `const byStructureItem = ( courseStructureItem ) => ( block )` instead? As in the line `const { name, attributes } = courseStructureItem;`, "attributes" is actually a property of block, not structure item. Also when we are calling this function like this `blocks.find( byStructureItem( structureItem ) );` below, we are passing the `structureItem` as the `block` param of `byStructureItem` function, and then `.find` function is calling the inner curried function with the iterating block, passing it as `courseStructureItem`.

Reply:
@Imran92 Very nice point!

Reply:
Ops, let me try another thing